tensorflow/compiler/mlir/hlo/include/mlir-hlo/Dialect/mhlo/IR/hlo_ops.td

```diff
@@ -1227,8 +1228,12 @@ def HLO_BatchNormTrainingOp : HLO_Op<"batch_norm_training", [NoSideEffect]> {
     HLO_Tensor:$operand,
     HLO_Tensor:$scale,
     HLO_Tensor:$offset,
+    HLO_Tensor:$side_input,
     F32Attr:$epsilon,
-    I64Attr:$feature_index
+    I64Attr:$feature_index,
+    I64Attr:$reserve_space_size,
+    BoolAttr:$use_reserve_space,
```

Comment:
I could be missing something here, but we don't allow backend specific concepts at the HLO layer.  Can you do this optimization in a way that does not require adding these inputs to BatchNorm?

Reply:
The additional attributes introduced here are optional and the HLO will should work with all backends. In this implementation, I am leveraging the TensorFlow BN op's additional reserve space output and the data flow is based on the correct lowering of graph-def at the bridge. Incase there is no side_input and/or activation, the HLO for BatchNorm remains the same.